core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1475,6 +1515,8 @@ class Log(@volatile var dir: File,
    * This allows reads to happen concurrently without synchronization and without the possibility of physically
    * deleting a file while it is being read from.
    *
+   * This method does not need to convert IOException to KafkaStorageException because it is only called before all logs are loaded
```

Comment:
This method is actually called in places after log loading. But in all other places, IOExceptions are handled by the callers. Perhaps we can adjust the comment a bit.